### PR TITLE
Populate mentor_user_id for new ParticipantDeclarations

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-2253](https://dfedigital.atlassian.net/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-2253)

### Context

The `ParticipantDeclaration` has a new foreign key `mentor_user_id` that is currently unpopulated. We want to
populate this with the induction record relative to the `declaration_date` at the time of creation.

### Changes proposed in this pull request

- Populate mentor_user_id for new declarations
- Clarify induction record date edge cases

### Guidance to review

The field is not yet referenced/read anywhere. We will backfill the existing declarations in a separate PR.

We have induction records in the database that are in an invalid/unexpected state. Until we are able to clean up the
data and ensure consistency we need to make sure the latest induction record logic is expected/documented.

Update `InductionRecord` scopes to prioritise records based on the latest `end_date` if they have the same `start_date`.

Add test coverage to the `FindBy` class to cover the following edge cases:

- Induction records with the same start date where one has a `nil` end date
- Induction records with the same start date where one has a later end date
- The expected scenario of the latest induction record having a `nil` end date

### CODE REVIEW

File: app/services/record_declaration.rb [hunk 1]
```diff
@@ -156,10 +156,23 @@ def declaration_parameters
...
     }
   end
 
+  def mentor
+    return nil unless participant_profile.ect?
+
+    induction_record = Induction::FindBy.call(
+      participant_profile:,
+      lead_provider: cpd_lead_provider.lead_provider,
+      date_range: (Date.new(0)..declaration_date),
```
Comment (peteryates) [inline app/services/record_declaration.rb:170]: not that it matters here but would `..declaration_date` make sense here instead of the zeroed date?

**Comment by ethax-ross** [comment]
The review app database is in a funny state due to changing the migrations under its feet; I'm going to close the PR to destroy the review apps and re-open it to create them fresh.

File: app/services/record_declaration.rb [hunk 2]
```diff
@@ -156,10 +156,23 @@ def declaration_parameters
       declaration_type:,
       cpd_lead_provider:,
       delivery_partner:,
+      mentor_user_id: mentor&.id,
       user: participant_identity.user,
     }
   end
 
+  def mentor
```
Comment (cwrw) [inline app/services/record_declaration.rb:164]: suggestion: could we name this `mentor_user` to avoid confusing it with a profile?

File: app/services/record_declaration.rb [hunk 3]
```diff
@@ -156,10 +156,23 @@ def declaration_parameters
...
+    return nil unless participant_profile.ect?
+
+    induction_record = Induction::FindBy.call(
```
Comment (cwrw) [inline app/services/record_declaration.rb:167]: having a deeper think about this, we probably need the time range between the start date AND end date of an induction record, if the end date is empty it means that one is the latest (in some cases 😅). Let's chat tomorrow

File: app/services/record_declaration.rb (hunk 2 above)
Comment (ethax-ross) [reply app/services/record_declaration.rb:164]: I'll rename to `mentor_user_id` as it looks like I can drop the `.user` access and just grab `user_id` of the participant identity

**Review by cwrw — APPROVED** [review]
